package/base-files/files/lib/upgrade/stage2

```diff
@@ -123,15 +123,22 @@ kill_remaining() { # [ <signal> [ <loop> ] ]
 
 indicate_upgrade
 
+/etc/init.d/dropbear stop
```

Comment:
Since calling init scripts can have unknown side effects (you don't know what other actions happen inside the "stop" implementation) it might be better to to invoke lower level procd procedures to terminate the services. Instead of invoking the init scripts, I suggest the following:

```
ubus call service delete '{ "name": "dropbear" }'
```

While being at it, you should also condense these different calls into a list of service names and a loop iterating the list:
```bash
stop_services="dropbear odhcpd rpcd sysntpd uhttpd wpad"

for service in $stop_services; do
  ubus call service delete '{ "name": "'"$service"'" }'   # mind the quoting
done
```

Even better would be stopping *all* procd services instead of just a known subset:

```bash
eval $(ubus call service list | jsonfilter -e 'running_services=@')     # this might require adding jsonfilter to the ramfs list

for service in $running_services; do
  ubus call service delete '{ "name": "'"$service"'" }'   # mind the quoting
done
```

Reply:
thanks for the examples, I will try that

I believe this is what @wulfy23 was talking about but I'm not sure exactly what was meant

Reply:
most of the time neither do I ;) ... most of my input is purely food for thought... and if nothing hits home can safely be ignored :)